src/include/optimizer/optimizer.h

```diff
@@ -85,16 +88,18 @@ class Optimizer : public AbstractOptimizer {
 
   void Reset() override;
 
-  OptimizerMetadata &GetMetadata() { return metadata_; }
+  OptimizerMetadata<Operator,OpType,OperatorExpression> &GetMetadata() { return metadata_; }
 
   /* For test purposes only */
```

Comment:
Kind of inherited from bad decisions before, but in terrier we try pretty hard to not have public test-only functions and use FRIEND_TEST instead.